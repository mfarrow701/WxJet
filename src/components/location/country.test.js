import React from 'react';
import {shallow} from 'enzyme';
import Country from './country';

describe('Country component', () => {
    let component;

    const renderWrapper = () => {
        component = shallow(<Country/>);
    };

    beforeEach(() => {
        renderWrapper();
    });

    it('should render the country component after load', () => {
        expect(component).toMatchSnapshot();
        expect(component.find('.Country').text()).toEqual('No country specified');
    });

    it('should render the country component with a custom message', () => {
        component.setProps({value: 'United Kingdom'});
        expect(component.find('.Country').text()).toEqual('United Kingdom');
    });
});
