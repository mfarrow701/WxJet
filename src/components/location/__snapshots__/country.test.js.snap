// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component should render the country component after load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Country />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "No country specified",
      "className": "Country",
    },
    "ref": null,
    "rendered": "No country specified",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No country specified",
        "className": "Country",
      },
      "ref": null,
      "rendered": "No country specified",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
